### PR TITLE
docs: Document Dex setup

### LINKED ISSUE #1786: Make it easier for users to set up and use OIDC
**Short description**
As a user who wants to configure OIDC and some static users, I can find clear docs and maybe some templates which let me do this with minimal know-how and fiddling about.

**Acceptance criteria**
2 parts to this.

As a user who wants to configure OIDC:
- [ ] I can find very clear and helpful docs/templates on how to set up Dex in my cluster
- [ ] I can navigate to (or be redirected to) the wego /sign_in page
- [ ] I can click the OIDC link and be directed to Dex
- [ ] I can see an option to login with Whatever the sso is
- [ ] I am redirected to my wego dash

As a user who wants to configure static users:
- [ ] I can find very clear and helpful docs/templates on how to set up Dex with static users in my cluster
- [ ] I can configure some static users on my cluster
- [ ] I can navigate to (or be redirected to) the wego /sign_in page
- [ ] I can click the OIDC link and be directed to Dex
- [ ] I can see an option to login with email/password combo
- [ ] I can fill out the form
- [ ] I am redirected to my wego dash

### PR DESCRIPTION
**Pull request by yitsushi**
Closes #1786

### CODE REVIEW

File: doc/setting-up-dex.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+[Dex][tool-dex] is an identity service that uses [OpenID Connect][oidc] to
+drive authentication for other apps.
+
+[tool-dex]: asd
+[oidc]: asd
+
+## Boostrap Flux
+
+TODO
```
Comment (Callisto13) [inline doc/setting-up-dex.md:13]: is this coming now or later?

File: doc/setting-up-dex.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+## Add cert-manager
+
+First install `cert-manager` as an TLS certificates is required to use Dex, the
+easiest way is to install `cert-manager`.
+
+_If you have other solution like [Traefik][tool-traefik] or
```
Comment (Callisto13) [inline doc/setting-up-dex.md:20]:
```suggestion
_If you have another solution like [Traefik][tool-traefik] or
```

File: doc/setting-up-dex.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+```yaml
+orgs:
+- name: weaveworks
+  teams:
+  - quick-silver
+  - mauvelous
+  - QA
+```
+
+Here we can defined groups under a GitHub organisation. In this example the
```
Comment (Callisto13) [inline doc/setting-up-dex.md:218]:
```suggestion
Here we can define groups under a GitHub organisation. In this example the
```

**Review by Callisto13 — COMMENTED** [review]
more coming later, have to 🏃‍♀️

File: doc/setting-up-dex.md (hunk 1 above)
Comment (yitsushi) [reply doc/setting-up-dex.md:13]: I think this file meant to be deleted as we did the documentation under `website/`

Comment (yitsushi) [reply doc/setting-up-dex.md:13]: Will remove the `Boostrap Flux` and add it as a pre-requirement

File: doc/setting-up-dex.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+Create a namespace where Dex will live:
+
+```yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: dex
+```
+
```
[the remark this answers is not in the document]
Comment (yitsushi) [reply doc/setting-up-dex.md:94]: well... they can apply them without Flux is they want. 🤷‍♀️

[the remark this answers is not in the document]
Comment (Callisto13) [reply doc/setting-up-dex.md:96]: true true

File: website/docs/guides/setting-up-dex.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+## Add credentials
+
+There are a [lot of options][dex-connectors] available with Dex, in this guide we will
+use the [GitHub connector][dex-github].
+
+We can get a GitHub ClientID and Client secret by creating a
+[new OAuth appliation][github-oauth].
+
+TODO - should we provide more detail for the above?
```
Comment (Callisto13) [inline website/docs/guides/setting-up-dex.md:45]: TODO? which above?

File: website/docs/guides/setting-up-dex.md [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+```yaml
+orgs:
+- name: weaveworks
+  teams:
+  - team-a
+  - team-b
+  - QA
+```
+
+Here we can defined groups under a GitHub organisation. In this example the
```
Comment (Callisto13) [inline website/docs/guides/setting-up-dex.md:177]:
```suggestion
Here we can define groups under a GitHub organisation. In this example the
```

File: website/docs/guides/setting-up-dex.md [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+        userID: "08a8684b-db88-4b73-90a9-3cd1661f5466"
+```
+
+Static user password can be generated with `htpasswd`:
+
+```bash
+echo password | htpasswd -BinC 10 admin | cut -d: -f2
+```
+
+TODO add completion screens
```
Comment (Callisto13) [inline website/docs/guides/setting-up-dex.md:265]: TODO

**Review by Callisto13 — APPROVED** [review]
All looks good 👍 
I am gonna run through a setup (I have never used Dex so it will be useful)

File: website/docs/guides/setting-up-dex.md (hunk 5 above)
Comment (yitsushi) [reply website/docs/guides/setting-up-dex.md:45]: /summon @sympatheticmoose

Comment (yitsushi) [reply website/docs/guides/setting-up-dex.md:45]: i guess i'll just attach an image about the oauth app

File: website/docs/guides/setting-up-dex.md [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+  name: dex
+```
+
+## Add credentials
+
+There are a [lot of options][dex-connectors] available with Dex, in this guide we will
+use the [GitHub connector][dex-github].
+
+We can get a GitHub ClientID and Client secret by creating a
+[new OAuth appliation][github-oauth].
```
[the remark this answers is not in the document]
Comment (yitsushi) [reply website/docs/guides/setting-up-dex.md:43]: I think the image show very clearly that part

File: website/docs/guides/setting-up-dex.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
+  apiGroup: rbac.authorization.k8s.io
+```
+
+### Set up static user
+
+For static user, add `staticPasswords` to the `config`:
```
Comment (Callisto13) [inline website/docs/guides/setting-up-dex.md:246]: this is already in the config. should we mention the inverse instead?

Comment (yitsushi) [reply website/docs/guides/setting-up-dex.md:241]: It's not in there anymore.

**Comment by yitsushi** [comment]
Rebased from main (and squashed commits)

**Pull request merged**